### DISCUSSION TITLE
Liveboards with multiple lines in the description do not export properly in metadata files

### DISCUSSION THREAD

**Discussion by brianbiggs** [thoughtspot/cs_tools#129] [General]
Some of our ThoughtSpot users have built Answers or Liveboards that have multiple lines (with Carriage Returns) in the **_Description_** field. When exporting to `ts_dependent_object.csv` (and maybe other files as well), those descriptions end up splitting the CSV lines into multiple lines, messing up the export (and eventual import).

Has this been reported before? Could CS Tolls be modified to encode those CR's so that they export (and thus import) correctly? Is some other workaround available?

**Comment by boonhapus** [comment]
Hey @brianbiggs , this has been reported a couple times and I think it's primarily because most tools don't ask for/respect the quoting character by default. In the CSV syncer, we have the quoting strategy of "minimal" by default, which wraps any cell/value with a quote character (`"`) if the cell itself contains a newline or delimiter (`|` is default).

By default, the data should look like this, where only the `Description` value here is quoted.

```csv
cluster_guid|dependent_guid|column_guid|name|description|author_guid|created|modified|object_type|object_subtype|is_verified
...
57421458-3c3a-11eb-8bb7-0204e|15e2e38a-55da-42fa-8c14-249ffa3211a5|ff107040-1ae3-427c-a496-77510904e83e|Delta for Best Answer|"A
B
C
D
E
F
G"|27741c8c-cc2c-44b9-8a0f-803e770e319e|2024-04-26 12:41:59|2024-04-27 16:21:13|QUESTION_ANSWER_BOOK||
...
```

You can wrap all cells with the quote and enforce more uniformity by using `quoting=ALL` and then the auto-parsing of Excel/GoogleSheets _should_ notice this and enforce it... but that's not always the case.

`cs_tools tools searchable metadata --syncer csv://directory=...&quoting=ALL`

Which will yield tabular data like this..

```csv
cluster_guid|dependent_guid|column_guid|name|description|author_guid|created|modified|object_type|object_subtype|is_verified
...
"57421458-3c3a-11eb-8bb7-0204e"|"15e2e38a-55da-42fa-8c14-249ffa3211a5"|"ff107040-1ae3-427c-a496-77510904e83e"|"Delta for Best Answer"|"A
B
C
D
E
F
G"|"27741c8c-cc2c-44b9-8a0f-803e770e319e"|"2024-04-26 12:41:59"|"2024-04-27 16:21:13"|"QUESTION_ANSWER_BOOK"|""|""
...
```

**Reply by brianbiggs** [reply]
OK, I'll admit I didn't try to import it into any tools (Google, Excel, etc.) other than Notepad++. I was using some `grep` commands on the raw files and those don't work because the single record is split on multiple lines. What you described above should work for importing CSV files somewhere, and if I can get Searchable working (#128), this problem will be moot and I can just do my queries in Searchable. Thanks!
